JSDemos/Demos/Charts/BarSparklines/React/App.js

```diff
@@ -20,57 +19,53 @@ const dataSource = new DataSource({
   paginate: false,
 });
 
-class App extends React.Component {
-  constructor(props) {
-    super(props);
-    this.source = dataSource;
-    this.onValueChanged = this.onValueChanged.bind(this);
-  }
+function App() {
+  React.useEffect(() => {
```

Comment:
why do we need to do this if it's inside onvalueChange callback?

Reply:
Yes, it is strange. I will fix it